src/Tracing/Transaction.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry\Tracing;
+
+use Sentry\Event;
+use Sentry\State\HubInterface;
+
+/**
+ * This class stores all the information about a Transaction.
+ */
+final class Transaction extends Span
+{
+    /**
+     * @var HubInterface|null The hub instance
+     */
+    private $hub;
+
+    /**
+     * @var string Name of the transaction
+     */
+    private $name;
+
+    /**
+     * Span constructor.
+     *
+     * @param TransactionContext|null $context The context to create the transaction with
+     * @param HubInterface|null       $hub     Instance of a hub to flush the transaction
+     *
+     * @internal
+     */
+    public function __construct(?TransactionContext $context = null, ?HubInterface $hub = null)
+    {
+        parent::__construct($context);
+        $this->hub = $hub;
+
+        $this->name = $context->name ?? '<unlabled transaction>';
+    }
+
+    /**
+     * @param string $name Name of the transaction
+     */
+    public function setName(string $name): void
+    {
+        $this->name = $name;
+    }
+
+    /**
+     * @return array<string, mixed> Returns the trace context
+     */
+    public function getTraceContext(): array
+    {
+        $data = $this->toArray();
+        unset($data['start_timestamp']);
```

Comment:
Please add  blank line above this one to improve readability and context-switching